pkg/server/server.go

```diff
@@ -255,79 +255,79 @@ func (s *ProxyServer) getBackend(reqHost string) (Backend, error) {
 	return nil, &ErrNotFound{}
 }
 
-func (s *ProxyServer) addBackend(agentID string, conn agent.AgentService_ConnectServer) (backend Backend) {
+func (s *ProxyServer) addBackend(agentID string, backend Backend) {
 	for i := 0; i < len(s.BackendManagers); i++ {
 		switch s.BackendManagers[i].(type) {
 		case *DestHostBackendManager:
```

Comment:
bleh... this feels like an abstraction failure. I'd prefer to encapsulate this logic in the `BackendManager`, rather than switching on the type here.

The way I would do it is:
1. make the current `BackendManager.AddBackend(...)` method private
2. make a new `BackendManager.AddBackend(backend)` method that encapsulates the logic here: call the private version for all the relevant IDs. If the backend doesn't have IDs that are relevant for that BackendManager, then it's a no-op


Then, this method `proxyServer.addBackend` just becomes a simple loop through the backend managers

You don't need to address this in this PR if you don't want.

Reply:
opened https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/516